### ISSUE TITLE
Bug in the Fourier loss

### ISSUE THREAD

**Issue by dhassault** [eth-siplab/Learning-with-FrameProjections#6]
I was wondering if this:
https://github.com/eth-siplab/Learning-with-FrameProjections/blob/47c9667d9d73f2450bd12d9dce7ac553bdd79c63/trainer_SSL_LE.py#L683-L689

Should not be:
```python
 ft_mapped  = map_ft(time_emb) 
 cwt_mapped = map_cwt(time_emb) 
  
 # Compute separate MSE losses. 
 loss_ft = torch.nn.L1Loss()(ft_mapped, FT_emb) 
 loss_cwt = torch.nn.L1Loss()(cwt_mapped, cwt_emb) 
```

With `loss_ft` being he difference between the predicted Fourier embedding by `map_ft` (`ft_mapped`) to the Fourier embedding from the Fourier encoder.

**Comment by Berken-demirel** [comment]
Hi @dhassault ,
Sorry for the late response. Yes, you’re right. That looks like I forgot to switch the target while doing some ablations.

If `map_ft` is meant to predict the Fourier embedding, then `loss_ft` should compare `ft_mapped` against the Fourier encoder output (`FT_emb`), not against `time_emb`. Comparing to `time_emb` would only make sense if we intentionally wanted `map_ft` to reconstruct the time embedding.

So the corrected code is:

```python
ft_mapped  = map_ft(time_emb)
cwt_mapped = map_cwt(time_emb)

# Compute separate L1 losses.
loss_ft  = torch.nn.L1Loss()(ft_mapped, FT_emb)
loss_cwt = torch.nn.L1Loss()(cwt_mapped, cwt_emb)
```

I’ve pushed a fix. It should be resolved now.
